### PR TITLE
Obscure Examine Panel Pref

### PR DESCRIPTION
**Pull request by GreytidePanda**
## About The Pull Request
Requested by several users on Discord

Adds a new game preference to toggle examine panel obscurity. Some users would still like their examine panel to be shown while they are 'unknown' (i.e. wearing a mask). The preference is obscurity on by default, so noone will be caught out by this.

I could have gone further with edits to OOC note visibility or pronouns etc, but don't want to meddle too much and this is as far as people seem to complain about it.

Preference On:

Preference Off:

## Changelog

:cl:
add: New preference for examine panel obscurity.
/:cl:

### CODE REVIEW

File: tgstation.dme [hunk 1]
```diff
@@ -7474,5 +7474,6 @@
 #include "modular_zubbers\maps\biodome\turfs.dm"
 #include "modular_zubbers\maps\biodome\vendcation.dm"
 #include "modular_zubbers\maps\biodome\weapons.dm"
+#include "modular_zubbers\master_files\code\modules\client\preferences\obscurity.dm"
```
Comment (nevimer) [inline tgstation.dme:7477]:
```suggestion
#include "modular_zubbers\master_files\code\modules\client\preferences\obscurity_examine.dm"
```

**Comment by ReturnToZender** [comment]
Good change (when it works), would be nice if it had a TGUI button toggle

**Comment by Cyprex** [comment]
Thanks a lot. Could you also make ghosts always able to see full FT etc. i. e. ghosts bypass obscurity.

**Comment by Cyprex** [comment]
But honestly I think the better option might be to have this be opt-out, and to have it be obscure, but if the user allows it there's a button "show unobscured" in the FT examine. With a big disclaimer that if this is abused to meta someones identity admins will ban you.

**Comment by nevimer** [comment]
It's good but it need comments before I can merge it

File: code/modules/mob/living/carbon/human/examine.dm [hunk 2]
```diff
@@ -498,7 +498,8 @@
 	/// The first 1-FLAVOR_PREVIEW_LIMIT characters in the mob's "flavor_text" DNA feature. FLAVOR_PREVIEW_LIMIT is defined in flavor_defines.dm.
 	var/preview_text = copytext_char((dna.features["flavor_text"]), 1, FLAVOR_PREVIEW_LIMIT)
 	// What examine_tgui.dm uses to determine if flavor text appears as "Obscured".
-	var/face_obscured = (wear_mask && (wear_mask.flags_inv & HIDEFACE)) || (head && (head.flags_inv & HIDEFACE))
+	var/obscurity_examine_pref = (client.prefs.read_preference(/datum/preference/toggle/obscurity_examine)) //bubberstation edit
+	var/face_obscured = (wear_mask && (wear_mask.flags_inv & HIDEFACE) && obscurity_examine_pref) || (head && (head.flags_inv & HIDEFACE) && obscurity_examine_pref)
```
Comment (nevimer) [inline code/modules/mob/living/carbon/human/examine.dm:502]:
```suggestion
	var/obscurity_examine_pref = (client.prefs.read_preference(/datum/preference/toggle/obscurity_examine)) //BUBBERSTATION EDIT
	var/face_obscured = (wear_mask && (wear_mask.flags_inv & HIDEFACE) && obscurity_examine_pref) || (head && (head.flags_inv & HIDEFACE) && obscurity_examine_pref) // BUBBERSTATION EDIT
```

File: modular_skyrat/master_files/code/modules/mob/living/examine_tgui.dm [hunk 3]
```diff
@@ -79,7 +80,7 @@
 
 	if(ishuman(holder))
 		var/mob/living/carbon/human/holder_human = holder
-		obscured = (holder_human.wear_mask && (holder_human.wear_mask.flags_inv & HIDEFACE)) || (holder_human.head && (holder_human.head.flags_inv & HIDEFACE))
+		obscured = (holder_human.wear_mask && (holder_human.wear_mask.flags_inv & HIDEFACE)) && obscurity_examine_pref || (holder_human.head && (holder_human.head.flags_inv & HIDEFACE) && obscurity_examine_pref)
```
Comment (nevimer) [inline modular_skyrat/master_files/code/modules/mob/living/examine_tgui.dm:83]:
```suggestion
		obscured = (holder_human.wear_mask && (holder_human.wear_mask.flags_inv & HIDEFACE)) && obscurity_examine_pref || (holder_human.head && (holder_human.head.flags_inv & HIDEFACE) && obscurity_examine_pref) // BUBBERSTATION EDIT - EXAMINE PREFS
```

**Review by nevimer — APPROVED** [review]

**Pull request merged**